### DISCUSSION TITLE
how to listen to real-time events on Sui

### DISCUSSION THREAD

**Discussion by NgocChung1998** [MystenLabs/sui#19494] [Q&A]
Let me ask, how can I listen to real-time events on Sui? The documentation for Sui events is no longer usable. Is there an alternative method I can use? Thank you, everyone! I use nodejs
let unsubscribe = await client.subscribeEvent({
filter: {
MovePackagePublished: {
package: Package,
},
},
onMessage: (event) => {
console.log("subscribeEvent", JSON.stringify(event, null, 2));
},
});

**Comment by ghost** [comment]
我有点想法，就是能否优化SUI链钱包地址？变成一个可简短可容易记住的数字，用户无需关注钱包的地址，因为地址关联着用户在注册后自动生成，避免用户增加心里成本，可行？

**Comment by sueun-dev** [comment]
`client.subscribeEvent()` is still the right shape for live events.

The part that usually breaks is the endpoint, not the callback.

A few things to check:
- use an RPC provider that supports websocket subscriptions
- keep the filter small at first
- if you need something more reliable, poll `queryEvents` or checkpoints with a cursor and store the last cursor yourself

So I would test the same code against a known good websocket fullnode first.
If it works there, your code is fine and the problem is the RPC endpoint.

Also, if you need replay after a restart, do not rely on subscriptions alone. Use a cursor-based poller for catch-up.
